src/test/java/core/service/impl/FileReaderImplTest.java

```diff
@@ -0,0 +1,56 @@
+package core.service.impl;
+
+import core.service.ReaderService;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FileReaderImplTest {
+    private static final String NON_EXISTING_FILENAME = "RAW_filename.csv";
+    private static final String EMPTY_FILENAME = "";
+    private static final String CORRECT_INPUT = "src/test/resources/testFile.csv";
+    private static final String EMPTY_FILE = "src/test/resources/testEmpty.csv";
+    private static final String WRONG_INPUT = "src/test/resources/test_file_wrong_data.csv";
+    private ReaderService fileReader = new FileReaderImpl();
+
+    @Test
+    public void readFromFile_correctInput_Ok() {
+        List<String> expected = List.of("type,fruit,quantity",
+                "b,banana,20",
+                "b,apple,100",
+                "s,banana,100",
+                "p,banana,13",
+                "r,apple,10",
+                "p,apple,20",
+                "p,banana,5",
+                "s,banana,50");
+        List<String> actual = fileReader.readFromFile(CORRECT_INPUT);
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void readFromFile_wrongData_Ok() {
+        List<String> expected = List.of("type,fruit,quantity // Something wrong!!!",
+                "b,banana,20",
+                "b,apple,100",
+                "s,banana,100");
+        List<String> actual = fileReader.readFromFile(WRONG_INPUT);
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void readFromFile_nonExistingFile_notOk() {
+        fileReader.readFromFile(NON_EXISTING_FILENAME);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void readFromFile_emptyFilename_notOk() {
+        fileReader.readFromFile(EMPTY_FILENAME);
+    }
+
+    @Test
+    public void readFromFile_readEmptyFile_Ok() {
+        List<String> emptyList = fileReader.readFromFile(EMPTY_FILE);
```

Comment:
It would be better to rename this variable because it seems like it always should be empty, but then you check it